### DISCUSSION TITLE
Process GET REST data

### DISCUSSION THREAD

**Discussion by joeytzy** [Budibase/budibase#12689] [Help] [answered]
So I have a GET rest request to retrieve data and I set it to retrieve an array objects of the data instead of the whole json data.

![image](https://github.com/Budibase/budibase/assets/44018390/62a34e1d-5a0d-49a3-a86e-87e2b7025811)

My question is, can I process the year and month into one single date and use the date to sort it in the data provider as well as the charts in the screen.

![image](https://github.com/Budibase/budibase/assets/44018390/a1a5e3be-4449-4cbd-92d9-909f1b9ee80c)

The above is my json data, the year is 2022 and the month is 9, how do I transform it to September 2022/ 9.22

**Comment by melohagan** [comment, marked as answer]
Hey @joeytzy 

As this is just JavaScript, you could probably find some examples online. Your transformer may look something like this:

```js
function getMonthName(monthNumber) {
  const date = new Date();
  date.setMonth(monthNumber - 1);

  return date.toLocaleString('en-US', {
    month: 'long',
  });
}

return data.yearly.sales.map(sale => ({ ...sale, label: getMonthName(sale.month) + " " + sale.year }))
```

reference: https://bobbyhadz.com/blog/javascript-convert-month-number-to-name

**Reply by joeytzy** [reply]
Ive modified parts of the code to ensure no error returned if the json data is empty. Otherwise, this works as intended. Thanks
